### DISCUSSION TITLE
add getters param to arrow fn style getters?

### DISCUSSION THREAD

**Discussion by theoephraim** [vuejs/pinia#1711] [Ideas]
In vuex, we used to define getters as `(state, getters) => ...`

Being able to just use `this` instead to compose getters is nice, but having the 2 different styles
- `arrowCounterPlusOne: (state) => state.counter + 1`
- `nonArrowCounterPlusOne(): number { return this.counter + 1 }`

is a bit weird, and the extra requirement that non-arrow getters must explicitly be typed is a bit annoying. Particularly if you are just shifting some data around (ie converting what was in state into a getter) you must then refactor any getters and add explicit types which can be quite complex...

I'm not sure if it's possible, but if we added back a `getters` param to the arrow-style getters, would we be able to skip the need for explicit typing? It seems perfectly doable from the JS side of things, but the TS types are quite complex...

**Comment by posva** [comment]
Unfortunately this cannot he typed (circular reference), that’s why we use `this`

**Reply by theoephraim** [reply]
shoot - that's too bad :(
Thanks for responding though.
